config/src/main/java/com/alibaba/nacos/config/server/service/repository/extrnal/ExternalStoragePersistServiceImpl.java

```diff
@@ -452,19 +476,25 @@ public boolean insertOrUpdateCas(String srcIp, String srcUser, ConfigInfo config
     public boolean insertOrUpdateCas(String srcIp, String srcUser, ConfigInfo configInfo, Timestamp time,
             Map<String, Object> configAdvanceInfo, boolean notify) {
         try {
-            addConfigInfo(srcIp, srcUser, configInfo, time, configAdvanceInfo, notify);
+            boolean updateResult = updateConfigInfoCas(configInfo, srcIp, srcUser, time, configAdvanceInfo, notify);
+            if (!updateResult) {
+                return addConfigInfo(srcIp, srcUser, configInfo, time, configAdvanceInfo, notify);
+            }
             return true;
-        } catch (DataIntegrityViolationException ive) { // Unique constraint conflict
-            return updateConfigInfoCas(configInfo, srcIp, srcUser, time, configAdvanceInfo, notify);
+        } catch (EmptyResultDataAccessException ex) { // No data, insert
+            return addConfigInfo(srcIp, srcUser, configInfo, time, configAdvanceInfo, notify);
```

Comment:
is there any cases that will throw EmptyResultDataAccessException in an update & add operation? same whit above.